### ISSUE TITLE
YouTube Revanced

### ISSUE THREAD

**Issue by mikheys** [ilyhalight/voice-over-translation#187]
Былоб круто, еслиб эта функция была в YouTube Revanced ☺️

**Comment by ilyhalight** [comment]
Было бы славно, но сомневаюсь, что когда-нибудь появится

**Comment by Woukim** [comment]
> Было бы славно, но сомневаюсь, что когда-нибудь появится

а это нереализуемо или почему не появится? это очень было бы востребовано...

**Comment by ilyhalight** [comment]
Чтобы это сделано нужно знать Kotlin

**Comment by Dragoy** [comment]
> Чтобы это сделано нужно знать Kotlin

подождем немного когда его будет знать в достаточной степени ChatGPT :>

**Comment by Metadorius** [comment]
Я засабмитил в репо РеВанседа, возможно, кто-то заинтересуется.

**Comment by LazyOverlordZindarak** [comment]
К сожалению, Revanced не лучший вариант. Это же модификация официального клиента. От сюда сложности со всяким дизассемблированием и модификациями. Исходный код приложения не открыт. От сюда и сложности. В первую очередь проще имплементировать технологию в какой-то open source клиент. А дальше, другие open source клиенты её имплементируют и проприетарные клиенты сделают. Когда появится пример, тогда разрабы revanced-а  скорее всего это тоже реализуют.

Я бы предложил NewPipe. Но его сейчас активно переписывают и issue новых фич удаляются.

**Comment by Metadorius** [comment]
Тем не менее, вансед, наверно, один из самых популярных клиентов. А насчёт "сложностей" - декомпиляция байт-кода, уж извините, не идёт особо в сравнение с дизассемблированием нативного кода, тем не менее, у меня есть собственный проект такого рода и контрибьюторы, которые этим занимаются. Было бы желание.

**Comment by LazyOverlordZindarak** [comment]
> Тем не менее, (ре)вансед, наверно, один из самых популярных клиентов. А насчёт "сложностей" - декомпиляция байт-кода, уж извините, не идёт особо в сравнение с дизассемблированием нативного кода, тем не менее, у меня есть собственный проект такого рода и контрибьюторы, которые этим занимаются. Было бы желание.

Revanced - vs - NewPipe
4.4k stars - vs - 25.3k stars
400 forks - vs - 2.8k forks

**Comment by LazyOverlordZindarak** [comment]
Есть какая-нибудь статистика популярности чтобы говорить что revanced самый популярный? "у меня есть собственный проект такого рода и контрибьюторы" - Интересно. Что за проект?

**Comment by LazyOverlordZindarak** [comment]
> Тем не менее, (ре)вансед, наверно, один из самых популярных клиентов. А насчёт "сложностей" - декомпиляция байт-кода, уж извините, не идёт особо в сравнение с дизассемблированием нативного кода, тем не менее, у меня есть собственный проект такого рода и контрибьюторы, которые этим занимаются. Было бы желание.


Есть какая-нибудь статистика популярности чтобы говорить что revanced самый популярный? "у меня есть собственный проект такого рода и контрибьюторы" - Интересно. Что за проект?

**Comment by Metadorius** [comment]
> Есть какая-нибудь статистика популярности чтобы говорить что revanced самый популярный? "у меня есть собственный проект такого рода и контрибьюторы" - Интересно. Что за проект?

Статистики нет, чисто субъективно. Сравнения звёзд и форков на гитхабе не совсем корректно, поскольку я веду речь и про вансед, и про ревансед, а ревансед появился сравнительно недавно.

Проект по расширению движка Red Alert 2 для исправления багов и мододельства. Можно в профиле поглядеть, Phobos  зовётся. У нас нет исходников, копаем дизассемблером и декомпилятором игру и изучаем.

> Я не говорю, что сделать это в revanced-е невозможно. Возможно. Но это на много сложнее чем просто добавить реализацию в open-source клиент. Легче и вероятнее сделать имплементацию для open source клиента, чем просто ждать пока кто-нибудь знающий сделает это в revanced или разбираться с декомпиляцией байт-кода и патчами самому. Нам нужен пример удачной реализации.

Можно и так, не спорю. Мне просто кажется, что нет желающих, вот и всё. Но тикет надо открыть другой под это, думается мне.

**Comment by Dragoy** [comment]
> Было бы ооочень здорово. У меня мама не понимает английского, смотрит только на мобильном с реванседа мной поставленного, очень порадовало бы.

Поставь просто маме для этого яндексовый браузер на телефон и пусть она там смотрит с переводом. Более нативного решения для мобилки пока нет.

**Comment by Metadorius** [comment]
> > Было бы ооочень здорово. У меня мама не понимает английского, смотрит только на мобильном с реванседа мной поставленного, очень порадовало бы.
> 
> Поставь просто маме для этого яндексовый браузер на телефон и пусть она там смотрит с переводом. Более нативного решения для мобилки пока нет. 

Проксирование нужно, я не в РФ живу. А впны с резалкой рекламы несовместимы. Рут - головняк.

**Comment by Dragoy** [comment]
> > > Было бы ооочень здорово. У меня мама не понимает английского, смотрит только на мобильном с реванседа мной поставленного, очень порадовало бы.
> > 
> > 
> > Поставь просто маме для этого яндексовый браузер на телефон и пусть она там смотрит с переводом. Более нативного решения для мобилки пока нет.
> 
> Проксирование нужно, я не в РФ живу. А впны с резалкой рекламы несовместимы. Рут - головняк.

Я тоже не в РФ и проблем никаких нет, первый раз слышу что для яндексовых сервисов кто-то впн юзает..

**Comment by Metadorius** [comment]
> Я тоже не в РФ и проблем никаких нет, первый раз слышу что для яндексовых сервисов кто-то впн юзает..

Ну это расширение как раз-таки и не работало без проксирования, пока не впилили его. Такие дела.

**Comment by LazyOverlordZindarak** [comment]
> > > Было бы ооочень здорово. У меня мама не понимает английского, смотрит только на мобильном с реванседа мной поставленного, очень порадовало бы.
> > 
> > 
> > Поставь просто маме для этого яндексовый браузер на телефон и пусть она там смотрит с переводом. Более нативного решения для мобилки пока нет.
> 
> Проксирование нужно, я не в РФ живу. А впны с резалкой рекламы несовместимы. Рут - головняк.

А как нативное приложение относится к проксированию? Я думал, ты за натив потому что это в разы удобнее.

На счёт проксирования, многие VPN приложения имеют функцию белого списка для приложений. То есть, можно добавить в белый список только яндекс браузер и тогда только его трафик пойдёт через VPN.

Есть ещё kiwi браузер. Это мобильный хром в который можно ставить расширения от настольного хрома. В том, числе этот сторонний VOT и VPN. Но для kiwi браузера это пожалуй тяжеловато.

**Comment by LazyOverlordZindarak** [comment]
Смотрю опен сорсные альтернативы ютуб клиента. Libre Tube на удивление хорош. Функций больше чем в Revanced.

**Comment by Spp595** [comment]
> Смотрю опен сорсные альтернативы ютуб клиента. Libre Tube на удивление хорош. Функций больше чем в Revanced.
> 
> В принципе, там уже есть функция аудиодорожек. Так что, имплементировать ai VOT будет относительно не сложно. Но надо разбираться. В любом случае, имплементировать фичу в опен сорс приложении несравненно проще чем копаться в патчинге revanced-а. И после того как будет сделана имплементация в опен сорс приложение, по идее будет проще сделать это в других, в том числе в revanced-е.

Согласен! Libretube достаточно хорош хотя все ещё не идеален. Возможно внедрить в него перевод будет легче

**Comment by NoPlagiarism** [comment]
> Было бы славно, но сомневаюсь, что когда-нибудь появится

А тогда смысл держать этот оффтопик открытым? Может лучше в discussions перенести хотя бы
